### PR TITLE
Refactor llm cache and fix bug

### LINKED ISSUE #1781: llm-cache: remove the usage of `using namespace vineyard` (except for tests cases) and put the class implementation in `namespace vineyard {}`.
e.g., https://github.com/v6d-io/v6d/blob/main/modules/llm-cache/radix-tree/radix-tree.cc#L24

Globally using namespace is bad as it pollutes a lots of contexts and is irreversible.

### LINKED ISSUE #1783: Pull types and type definitions into `namespace vineyard`
e.g., https://github.com/v6d-io/v6d/blob/main/modules/llm-cache/ds/kv_state_cache_block.h#L32-L55

### LINKED ISSUE #1788: llm-cache: too many warnings under clang/clang++ (on MacOS)
image

image

image

image

### PR DESCRIPTION
**Pull request by vegetableysm**
What do these changes do?
-------------------------

- Remove "using namespace vineyard" sentence.
- Put type definition to vineyard namespace.
- Fix compile errors and warnings on mac os m1 platform.
- Add radix tree to vineyard namespace.

Related issue number
--------------------

- Fixes #1781
- Fixes #1783
- Fixes #1788

### CODE REVIEW

File: modules/llm-cache/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ file(GLOB VINEYARD_LLM_CACHE_SRCS "${CMAKE_CURRENT_SOURCE_DIR}"
 )
 
 add_library(vineyard_llm_cache ${VINEYARD_LLM_CACHE_SRCS})
-target_link_libraries(vineyard_llm_cache PUBLIC vineyard_client vineyard_basic)
+target_link_libraries(vineyard_llm_cache PUBLIC libzstd_static ${GLOG_LIBRARIES} vineyard_client vineyard_basic)
```
Comment (sighingnow) [inline modules/llm-cache/CMakeLists.txt:11]: Don't publicly link aginist `libzstd_static`. As this target is not installed and exported. Use

```
target_link_libraries(... PRIVATE ...)
```

File: modules/llm-cache/ds/kv_state_cache_block.h [hunk 2]
```diff
@@ -29,11 +29,26 @@ limitations under the License.
...
 #include "llm-cache/radix-tree/radix-tree.h"
 
+// Set the bit to 1, which means the resource is not being used
+#define FREE_BIT_RESOURCE(value, bit) ((value) |= (((uint64_t) 1) << (bit)))
+
+// Set the bit to 0, which means the resource is being used
+#define ACQUIRE_BIT_RESOURCE(value, bit) \
+  ((value) &= (~(((uint64_t) 1) << (bit))))
+
+#define DEFAULT_BLOCK_SIZE 64
```
Comment (sighingnow) [inline modules/llm-cache/ds/kv_state_cache_block.h:39]: Use a `constexpr`.

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: modules/llm-cache/CMakeLists.txt (hunk 1 above)
Comment (vegetableysm) [reply modules/llm-cache/CMakeLists.txt]: Done

File: modules/llm-cache/ds/kv_state_cache_block.h (hunk 2 above)
Comment (vegetableysm) [reply modules/llm-cache/ds/kv_state_cache_block.h]: OK.
Does “ACQUIRE_BIT_RESOURCE” need to be a constexpr function as well?

Comment (sighingnow) [reply modules/llm-cache/ds/kv_state_cache_block.h]: I'm afraid it cannot be a constexpr function, as it argument `value` cannot be a constant.

File: modules/llm-cache/CMakeLists.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ file(GLOB VINEYARD_LLM_CACHE_SRCS "${CMAKE_CURRENT_SOURCE_DIR}"
 )
 
 add_library(vineyard_llm_cache ${VINEYARD_LLM_CACHE_SRCS})
-target_link_libraries(vineyard_llm_cache PUBLIC vineyard_client vineyard_basic)
+target_link_libraries(vineyard_llm_cache PRIVATE libzstd_static vineyard_client vineyard_basic)
```
Comment (sighingnow) [inline modules/llm-cache/CMakeLists.txt:11]:
```suggestion
target_link_libraries(vineyard_llm_cache PRIVATE libzstd_static)
target_link_libraries(vineyard_llm_cache PUBLIC vineyard_client vineyard_basic)
```

**Pull request merged**